source/auth/auth.rst

```diff
@@ -453,10 +573,77 @@ Implementation
 Test Plan
 =========
 
-Tests have been defined in the associated files:
+Connection string tests have been defined in the associated files:
 
 * `Connection String <tests/connection-string.json>`_.
 
+---------------------------------------
+SCRAM-SHA-256 and mechanism negotiation
+---------------------------------------
+
+Testing SCRAM-SHA-256 requires server version 3.7.3 or later with
+``featureCompatibilityVersion`` of "4.0" or later.
+
+Drivers that allow specifying auth parameters in code as well as via
+connection string should test both for the test cases described below.
+
+Step 1
+------
+
+Create three test users, one with only SHA-1, one with only SHA-256 and one
+with both.  For example::
+
+    db.createUser({user: 'sha1', pwd: 'sha1', roles: ['root'], mechanisms: ['SCRAM-SHA-1']})
+    db.createUser({user: 'sha256', pwd: 'sha256', roles: ['root'], mechanisms: ['SCRAM-SHA-256']})
+    db.createUser({user: 'both', pwd: 'both', roles: ['root'], mechanisms: ['SCRAM-SHA-1', 'SCRAM-SHA-256']})
+
+Optional, but strongly encouraged: Create a SCRAM-SHA-256 user with a username
```

Comment:
Why optional?

Reply:
I was thinking that it's not technically necessary to demonstrate that the driver can do SCRAM-SHA-256 and negotiation and couples our testing to the server's (possibly changing) rules.  Perils of spec writing before the server is "done".